src/VisualStudio/Core/Def/Implementation/LanguageService/AbstractCreateServicesOnTextViewConnection.cs

```diff
@@ -47,18 +54,31 @@ void IWpfTextViewConnectionListener.SubjectBuffersDisconnected(IWpfTextView text
         /// <summary>
         /// Must be invoked from the UI thread.
         /// </summary>
-        private void CreateServices(string languageName)
+        private void CreateServicesOnUIThread()
         {
             var serviceTypeAssemblyQualifiedName = typeof(ISnippetInfoService).AssemblyQualifiedName;
             foreach (var languageService in _languageServices)
             {
                 if (languageService.Metadata.ServiceType == serviceTypeAssemblyQualifiedName &&
-                    languageService.Metadata.Language == languageName)
+                    languageService.Metadata.Language == _languageName)
                 {
                     _ = languageService.Value;
                     break;
                 }
             }
```

Comment:
❔ Can this entire method be reduced now that we have the workspace?

```
_workspace.Services.GetLanguageService<ISnippetInfoService>(_languageName);
```

Reply:
Not sure about this, but it seems getting LS from workspace could be more expensive so we might still want to avoid doing it on UI thread?